src/cart/crt_swim.c

```diff
@@ -544,16 +632,40 @@ int crt_swim_init(int crt_ctx_idx)
 
 	rc = crt_proto_register(&crt_swim_proto_fmt);
 	if (rc) {
-		D_ERROR("crt_proto_register() failed=%d\n", rc);
+		D_ERROR("crt_proto_register() failed "DF_RC"\n", DP_RC(rc));
 		D_GOTO(cleanup, rc);
 	}
 
 	rc = crt_register_progress_cb(crt_swim_progress_cb, crt_ctx_idx, NULL);
 	if (rc) {
-		D_ERROR("crt_register_progress_cb() failed=%d\n", rc);
+		D_ERROR("crt_register_progress_cb() failed "DF_RC"\n",
+			DP_RC(rc));
 		D_GOTO(cleanup, rc);
 	}
 
+	if (!d_fault_inject_is_enabled())
+		D_GOTO(out, rc);
+
+	crt_swim_should_fail = false; /* disabled by default */
+	crt_swim_fail_hlc = 0;
+	crt_swim_fail_delay = 10;
+	crt_swim_fail_id = SWIM_ID_INVALID;
+
+	/* Search the attr in inject yml first */
+	d_fa_swim_drop_rpc = d_fault_attr_lookup(CRT_SWIM_FAIL_DROP_RPC);
+	if (d_fa_swim_drop_rpc != NULL) {
+		D_ERROR("*** fa_swim_drop_rpc: id=%u/0x%x, "
+			"interval=%u, max="DF_U64", x=%u, y=%u, args='%s'\n",
+			d_fa_swim_drop_rpc->fa_id,
+			d_fa_swim_drop_rpc->fa_id,
+			d_fa_swim_drop_rpc->fa_interval,
+			d_fa_swim_drop_rpc->fa_max_faults,
+			d_fa_swim_drop_rpc->fa_probability_x,
+			d_fa_swim_drop_rpc->fa_probability_y,
+			d_fa_swim_drop_rpc->fa_argument);
+		if (d_fa_swim_drop_rpc->fa_argument != NULL)
```

Comment:
What if it is NULL? Would that be an error because the fault injection file isn't complete?

Reply:
If argument is NULL we will use default values which are set in lines 649-652.